File: ex34.rb
#encoding:utf-8
animals = ['bear', 'python', 'peacock', 'kangaroo', 'whale', 'platypus']
puts "数组的长度为："+animals.size.to_s
puts animals
animals[0]='peacock'
animals[1]='whale'
animals[2]='bear'
animals[3]='platypus'
animals[6]='monkey'


puts "The 1st animal is at 0 and is a " +animals[0]
puts "The 2se animal is at 1 and is a " +animals[1]
puts "The 3rd animal is at 2 and is a " +animals[2]
puts "The 4fo animal is at 3 and is a " +animals[3]

puts animals.first
puts animals.last

puts "数组的长度为："+animals.size.to_s
puts "第二个元素为："+animals[1]
puts "最后一个元素为："+animals[-1]

File: LearnPython/ex34.py
#coding=UTF-8
#序数ordinal==有序,以1开始;基数cardinal==随机选取,以0开始,序数-1==基数
animals = ['bear', 'python', 'peacock', 'kangaroo', 'whale', 'platypus']
print 'The animal at 1 is the 2nd animal and is a ',animals[1] 
print 'The 3rd animal is at 2 and is a ',animals[3-1]
print 'The 1st animal is at 0 and is a ',animals[1-1]
print 'The animal at 3 is the 4th animal and is a ',animals[3]
print 'The 5th animal is at 4 and is a',animals[5-1]
print 'The animal at 2 is the 3rd animal and is a',animals[2]
print 'The 6th animal is at 5 and is a',animals[6-1]
print 'The animal at 4 is the 5th animal and is a',animals[4]
assert animals[5] == 'platypus'
assert animals[4] == 'whale'